dplock/dplock.go

```diff
@@ -130,40 +130,44 @@ func (l *Lock) Acquire(ctx context.Context, id string) (lockID string, err error
 	for {
 		lockID, err = l.Lock(ctx, id)
 		if err != lock.ErrAlreadyLocked {
-			return lockID, err
+			return lockID, err // Successful or failed due to some generic error, no retry is attempted
 		}
 		if retries >= AcquireMaxRetries {
-			return "", ErrAcquireMaxRetries
+			return "", ErrAcquireMaxRetries // Failed too many times
 		}
 		retries++
 		select {
 		case <-time.After(AcquirePeriod):
-			continue
+			continue // Retry
 		case <-l.CloserChannel:
 			log.Info(ctx, "stop acquiring lock. Mongo db is being closed")
-			return "", ErrMongoDbClosing
+			return "", ErrMongoDbClosing // Abort because the app is closing
 		}
 	}
 }
 
 // Unlock releases an exclusive mongoDB lock for the provided id (if it exists)
-func (l *Lock) Unlock(ctx context.Context, lockID string) error {
+func (l *Lock) Unlock(ctx context.Context, lockID string) {
 	retries := 0
 	for {
 		_, err := l.Client.Unlock(ctx, lockID)
 		if err == nil {
-			return nil
+			if retries > 0 {
+				log.Info(ctx, "unlocking succeeded after some retries", log.Data{"retries": retries})
```

Comment:
I think a comment needs to be added for this log line to explain that this log is here on a temporary basis so that the log files can be examined and for the then determining if we are seeing unlock failures and retries working - so that we may try to understand why or what is causing the unlock failures elsewhere.
I also think the retry count should be increased to a value that allows the retries to keep hapenning for at least half a second.
This value can be reduced after we have inspected the log files for this error and the team has gotten a feel for what value is more suitable.
Thanks.